File: lighthouse/src/main/java/com/automation/center/lighthouse/model/MetricResult.java
package com.automation.center.lighthouse.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity(name = "metric_result")
public class MetricResult {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "metric_id")
    private Metric metric;
    @ManyToOne
    @JoinColumn(name = "metric_url_id")
    private MetricUrl metricUrl;
    private String value;
}

File: lighthouse/src/main/java/com/automation/center/lighthouse/model/MetricResult.kt
package com.automation.center.lighthouse.model

import jakarta.persistence.*
import java.time.LocalDateTime

@Table(name = "metric_result")
@Entity
class MetricResult() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    @ManyToOne
    @JoinColumn(name = "url_id")
    var url: Page? = null

    @ManyToOne
    @JoinColumn(name = "metric_id")
    var metric: Metric? = null

    @Column(name = "\"value\"")
    var value: String? = null

    @Column(name = "date")
    var date: LocalDateTime? = null

    constructor(id: Long) : this() {
        this.id = id
    }
}